### PR TITLE
feat: Add some eslint rules for typescript and React

### PR DESCRIPTION
**Pull request by tatsushitoji**
## TL;DR
- Remove removed `prefer-interface` rule
- Enable lint dotfile (.eslintrc.js only)
- Only lint for this repository is javascript
- Add some import rules
- Add some React rules

## check list
- [x] pass CI

### CODE REVIEW

File: packages/eslint-config-globis/index.js [hunk 1]
```diff
@@ -1,3 +1,13 @@
...
+  rules: {
+    'import/order': ['error', { 'newlines-between': 'always' }],
+    'import/prefer-default-export': 0,
+    // FixMe https://github.com/yannickcr/eslint-plugin-react/issues/1846
+    'react/button-has-type': 'off',
+    'react/jsx-filename-extension': [1, { extensions: ['.tsx'] }],
+    'react/jsx-key': ['error', { checkFragmentShorthand: true }],
+    'react/jsx-props-no-spreading': 0,
+    'react/prop-types': 0,
+  },
```
Comment (shoota) [inline packages/eslint-config-globis/index.js:12]: badge
To simplify, I prefer these rules are defined to another one.

Comment (tatsushitoji) [reply packages/eslint-config-globis/index.js:12]: I think it is a prerequisite configuration that uses React.

e.g) 
- https://github.com/globis-org/frontend-standard/blob/master/packages/eslint-config-globis/shared.js#L4
- https://github.com/globis-org/frontend-standard/blob/master/packages/eslint-config-globis/index.js#L2

Do we want to separate React configurations from the settings above?

Comment (shoota) [reply packages/eslint-config-globis/index.js:12]: hmmm... 
We should clarify each file duty, to have same context. 🤓
It seem strange a little for me,  to `base.js` extends shared.js.

Comment (tatsushitoji) [reply packages/eslint-config-globis/index.js:12]: > We should clarify each file duty, to have same context

+1

**Review by dkimura — APPROVED** [review]

**Pull request merged**